cassandra-executor/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraExecutor.java

```diff
@@ -93,77 +124,35 @@ public void launchTask(final ExecutorDriver driver, final TaskInfo task) {
                     driver.sendStatusUpdate(taskStatus(task, TaskState.TASK_RUNNING, details));
                     break;
                 case CASSANDRA_SERVER_RUN:
-                    lastKnownStatus = null;
-                    final Process cassandraProcess = launchCassandraNodeTask(taskIdMarker, taskDetails.getCassandraServerRunTask());
-                    process = cassandraProcess;
-                    serverTaskID = task.getTaskId();
-                    details = SlaveStatusDetails.newBuilder()
-                            .setStatusDetailsType(SlaveStatusDetails.StatusDetailsType.SERVER_RUN_DETAILS)
-                            .build();
-                    driver.sendStatusUpdate(taskStatus(task, TaskState.TASK_STARTING, details));
+                    safeShutdown();
+                    serverTask = task;
+                    serverTaskStatusOutstanding = true;
+                    process = objectFactory.launchCassandraNodeTask(taskIdMarker, taskDetails.getCassandraServerRunTask());
+                    jmxConnect = objectFactory.newJmxConnect(taskDetails.getCassandraServerRunTask().getJmx());
                     break;
                 case CASSANDRA_SERVER_SHUTDOWN:
-                    process.destroy();
-                    process = null;
+                    safeShutdown();
                     driver.sendStatusUpdate(taskStatus(task.getExecutor().getExecutorId(),
-                            serverTaskID,
+                            serverTask.getTaskId(),
                             TaskState.TASK_FINISHED,
                             nullSlaveStatusDetails()));
                     driver.sendStatusUpdate(taskStatus(task, TaskState.TASK_FINISHED));
                     break;
-                case CASSANDRA_SERVER_HEALTH_CHECK:
-                    final CassandraServerHealthCheckTask healthCheckTask = taskDetails.getCassandraServerHealthCheckTask();
+                case HEALTH_CHECK:
+                    final HealthCheckTask healthCheckTask = taskDetails.getHealthCheckTask();
                     LOGGER.info(taskIdMarker, "Received healthCheckTask: {}", protoToString(healthCheckTask));
-                    final CassandraNodeHealthCheckDetails healthCheck = performHealthCheck(healthCheckTask);
+                    final HealthCheckDetails healthCheck = performHealthCheck(driver);
```

Comment:
Please move `if (serverTaskStatusOutstanding) {...` from `performHealthCheck` out so that `performHealthCheck` doesn't have a dependency on the driver.